sdk/main/include/impl/OpenSSLObjectWrapper.h

```diff
@@ -0,0 +1,250 @@
+/*-
+ *
+ * Hedera C++ SDK
+ *
+ * Copyright (C) 2020 - 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#ifndef HEDERA_SDK_CPP_IMPL_OPENSSL_OBJECT_WRAPPER_H_
+#define HEDERA_SDK_CPP_IMPL_OPENSSL_OBJECT_WRAPPER_H_
+
+#include <memory>
+#include <openssl/decoder.h>
+#include <openssl/ec.h>
+#include <openssl/evp.h>
+
+namespace Hedera::internal
+{
+/**
+ * Templated base wrapper class to be used for OpenSSL objects that require custom deleter functions.
+ *
+ * @tparam ObjectType  The type of OpenSSL object this class should wrap.
+ * @tparam DeleterType The deleter type for the OpenSSL object.
+ */
+template<typename ObjectType, typename DeleterType>
+class OpenSSLObjectWrapper
+{
+public:
+  /**
+   * Get the wrapped OpenSSL object.
+   *
+   * @return A pointer to the wrapped OpenSSL object, nullptr if no object exists.
+   */
+  [[nodiscard]] ObjectType* get() const { return mObject.get(); }
+
+  /**
+   * Determine if this OpenSSLObjectWrapper has a valid OpenSSL object.
+   *
+   * @return \c TRUE if there exists an OpenSSL object, otherwise \c FALSE.
+   */
+  [[nodiscard]] explicit operator bool() const { return mObject != nullptr; }
+
+protected:
+  /**
+   * Construct with values for the object and its custom deleter.
+   *
+   * @param object  The OpenSSL object to wrap.
+   * @param deleter The custom deleter for the OpenSSL object.
+   */
+  OpenSSLObjectWrapper(ObjectType* object, DeleterType deleter)
+    : mObject({ object, deleter })
+  {
+  }
+
+private:
+  /**
+   * Pointer to the OpenSSL object with its associated deleter.
+   */
+  std::unique_ptr<ObjectType, DeleterType> mObject = { nullptr, DeleterType() };
+};
+
+/**
+ * Wrapper class for the OpenSSL ECDSA_SIG object.
+ */
+class OpenSSL_ECDSA_SIG : public OpenSSLObjectWrapper<ECDSA_SIG, decltype(&ECDSA_SIG_free)>
+{
+public:
+  /**
+   * Default constructor sets the ECDSA_SIG_free deleter function.
+   */
+  OpenSSL_ECDSA_SIG()
```

Comment:
Any reason we need a default constructor, rather than just deleting it?

Reply:
I suppose not, removed